internal/operator/internal/operator_installer.go

```diff
@@ -62,3 +66,67 @@ func (o OperatorInstaller) InstallOperator(ctx context.Context) (*v1alpha1.Clust
 
 	return todo, nil
 }
+
+func (o OperatorInstaller) verifyOperatorInstallation(ctx context.Context, ip *v1alpha1.InstallPlan) (*v1alpha1.ClusterServiceVersion, error) {
+	ipKey := types.NamespacedName{
+		Namespace: ip.GetNamespace(),
+		Name:      ip.GetName(),
+	}
+	if err := wait.PollImmediateUntil(time.Millisecond*250, func() (bool, error) {
+		if err := o.cfg.Client.Get(ctx, ipKey, ip); err != nil {
+			return false, err
+		}
+		if ip.Status.Phase == v1alpha1.InstallPlanPhaseComplete {
+			return true, nil
+		}
+		return false, nil
+	}, ctx.Done()); err != nil {
+		return nil, fmt.Errorf("waiting for operator installation to complete: %v", err)
+	}
+
+	// verify CSV
+	csv, err := o.verifyCSV(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("waiting for CSV installation to complete: %v", err)
+	}
+	return csv, nil
+}
+
+func (o OperatorInstaller) verifyCSV(ctx context.Context) (*v1alpha1.ClusterServiceVersion, error) {
+
+	csvKey := types.NamespacedName{
+		Namespace: o.cfg.Namespace,
+	}
+
+	if csvKey.Name == "" {
+		return nil, fmt.Errorf("could not find installed CSV in install plan")
+	}
+
+	csv := &v1alpha1.ClusterServiceVersion{}
+	if err := wait.PollImmediateUntil(time.Millisecond*250, func() (bool, error) {
+		if err := o.cfg.Client.Get(ctx, csvKey, csv); err != nil {
+			return false, fmt.Errorf("get clusterserviceversion: %v", err)
+		}
+		if csv.Status.Phase == v1alpha1.CSVPhaseSucceeded {
+			return true, nil
+		}
+		return false, nil
+	}, ctx.Done()); err != nil {
+		// Check deplpoyment status
+		dep := &appsv1.Deployment{}
+		depName := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Name
+		depKey := types.NamespacedName{
+			Namespace: csvKey.Namespace,
+			Name:      depName,
+		}
+		if err := o.cfg.Client.Get(ctx, depKey, dep); err != nil {
+			return nil, fmt.Errorf("get deployment: %v", err)
+		}
+		for _, s := range dep.Status.Conditions {
+			if s.Type != "Progressing" && s.Type != "Available" {
+				return nil, fmt.Errorf("Deployment failed: %v", s.Message)
+			}
+		}
+	}
```

Comment:
If the error is not nil, but we were unable to determine a root cause, we should still return the error, possibly wrapped with some extra context if we know anything else about it.